IBPSA/ThermalZones/ISO13790/BaseClasses/GlazedElements.mo

```diff
@@ -0,0 +1,301 @@
+  connect(corFacWest.y, fraWinWest.u)
+    annotation (Line(points={{61,-80},{70,-80}}, color={0,0,127}));
+  annotation (defaultComponentName="glaEle",Icon(coordinateSystem(preserveAspectRatio=false, extent={{-140,-140},
+            {140,140}}), graphics={
+        Rectangle(
+          extent={{-140,140},{140,-142}},
+          lineColor={95,95,95},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Rectangle(extent={{-140,140},{140,-142}}, lineColor={95,95,95}),
+        Polygon(
+          points={{20,100},{-20,80},{-20,-80},{20,-40},{20,100}},
+          lineColor={95,95,95},
+          fillColor={170,213,255},
+          fillPattern=FillPattern.Solid,
+          lineThickness=0.5),
+        Ellipse(
+          extent={{-74,128},{-124,78}},
+          lineColor={255,255,0},
+          lineThickness=0.5,
+          fillColor={244,125,35},
+          fillPattern=FillPattern.Sphere),
+        Text(
+          extent={{-110,198},{112,166}},
+          lineColor={0,0,255},
+          textString="%name")}),Diagram(coordinateSystem(preserveAspectRatio=
+            false, extent={{-140,-140},{140,140}})),
+         Documentation(info="<html>
+<p>
+This model calculates the solar heat gains through glazed elements. The heat flow by solar gains through building element k is given by
+<p align=\"center\" style=\"font-style:italic;\">
+&Phi;<sub>sol,k</sub> = F<sub>sh,ob,k</sub>A<sub>sol,k</sub>I<sub>sol,k</sub>-F<sub>r,k</sub>&Phi;<sub>r,k</sub>
+</p>
+where <i>F<sub>sh,ob,k</sub></i> is the shading reduction factor for external obstacles, 
+<i>A<sub>sol,k</sub></i> is the effective collecting area of surface k,
+<i>I<sub>sol,k</sub></i> is the solar irradiance per square meter, 
+<i>F<sub>r,k</sub></i> is the form factor between the building element and the sky, and 
+<i>&Phi;<sub>r,k</sub></i> is the extra heat flow due to thermal radiation to the sky. 
+The effective collecting area of glazed elements <i>A<sub>sol</sub></i> is calculated as
+<p align=\"center\" style=\"font-style:italic;\">
+A<sub>sol</sub> = F<sub>sh,gl</sub>F<sub>w</sub>g<sub>n</sub>(1-F<sub>f</sub>)A<sub>w</sub>
+</p>
+where <i>F<sub>sh,gl</sub></i> is the shading reduction factor for venetian blind or shades (equal to 1 in this model implementation),
+<i>F<sub>w</sub></i> is a correction factor (equal to 0.9 in this model implementation),
+<i>g<sub>n</sub></i> is the solar energy transmittance for radiation perpendicular to the window,
+<i>F<sub>f</sub></i> is the frame fraction, and
+<i>A<sub>w</sub></i> is the window area.
+In this model implementation, the extra radiative heat flow due to thermal radiation to the sky was neglected.
+
+
+
+</p>
+</html>",
+revisions="<html>
+<ul>
+<li>
+Mar 16, 2022, by Alessandro Maccarini:<br/>
+First implementation.
+</li>
+</ul>
+</html>",
+        info="<html>
+<p>
+Mass data for heavy building
+</p>
+</html>"));
+end GlazedElements;
```

Comment:
Good point. The thing is that at the moment I am not sure which variables could be useful for users. I probably need to use the models a bit more in order to figure out this aspect. I will keep this in mind for a future revision of the models. Do you have some specific examples of variables that could be protected?

Reply:
In this very model, I would for example declare the `sum`-model or `fraWinWest` as protected since it only sums up already given/reachable variables. Another example are constants (not in this very model).
But of course this could be optimised by using the models.